pymeasure/instruments/thyracont/transmitter.py

```diff
@@ -0,0 +1,491 @@
+    """
+    value = f"{value}"
+    return f"{len(value):02}{value}"
+
+
+source = {
+    'combination': 0,
+    'pirani': 1,
+    'piezo': 2,
+    'hotCathode': 3,
+    'coldCathode': 4,
+    'ambient': 6,
+    'relative': 7,
+}
+
+
+class Sources(IntEnum):
+    COMBINATION = 0
+    PIRANI = 1
+    PIEZO = 2
+    HOT_CATHODE = 3
+    COLD_CATHODE = 4
+    AMBIENT = 6
+    RELATIVE = 7
+
+
+gas_factor = Channel.control(
+    "0C{ch}00", "0C{ch}%s", "Control the gas correction factor.",
+    values=(0.2, 8),
+    validator=validators.strict_range,
+    set_process=compose_data,
+)
+
+
+class SensorChannel(Channel):
+    """Generic channel for individual pressure sensors of a Transmitter."""
+    _id = -1  # obligatory channel number, define in channel types
+
+    def __init__(self, parent, id):
+        if id is None:
+            id = self._id
+        assert id == self._id, f"Pirani ID has to be {self._id}."
+        super().__init__(parent, id=self._id)
+
+    pressure = Channel.measurement(
+        "0M{ch}00", """Get the current pressure in mbar.""",
+        preprocess_reply=lambda r: r.replace("UR", "0").replace("OR", "inf"),
+    )
+
+
+class Pirani(SensorChannel):
+    """Pirani sensor channel.
+
+    A Pirani sensor measures the heat transfer in the gas. The reading depends on the gas type.
+    The sensor reading is linear to the real pressure below some threshold, around 1 mbar.
+    You may control a gas factor with :attr:`gas_factor`.
+    """
+
+    _id = 1
+
+    gas_factor = gas_factor
+
+    statistics = Channel.measurement(
+        "0PM011",
+        """Get the sensor statistics: wear in percent (negative: corrosion,
+            positive: contamination), time since last adjustment in hours.""",
+        separator="A",
+        cast=int,
+        get_process=lambda vals: [v / 4 for v in vals],
+    )
+
+
+class Piezo(SensorChannel):
+    """Piezo sensor channel. A piezo sensor is independent of the gas present."""
+
+    _id = 2
+
+
+class HotCathode(SensorChannel):
+    """Hot cathode sensor channel."""
+
+    _id = 3
+
+    filament_mode = Instrument.control(
+        "0FC00", "2FC01%i", """The hot cathode filament control setting.""",
+        values={"2 if 1 defect": 0,
+                "Filament1": 1,
+                "Filament2": 2,
+                "toggle>1mbar": 3},
+        validator=validators.strict_discrete_set,
+        map_values=True,
+        cast=int,
+        check_set_errors=True,
+    )
+
+    active_filament = Instrument.measurement(
+        "0FN00", "Get the current filament number."
+    )
+
+    filament_status = Instrument.measurement(
+        "0FS00", """Get the status of the hot cathode filaments.""",
+        values=["Filament 1 and 2 ok",
+                "Filament 1 defective",
+                "Filament 2 defective",
+                "Filament 1 and 2 defective"],
+        map_values=True,
+    )
+
+    gas_factor = gas_factor
+
+    statistics = Channel.measurement(
```

Comment:
Note the code here is copy pasted from the Pirani channel but the manual says that the return value is different. For my VSH model this command unfortunately does not work at all. If I send it I get a 
`ConnectionError: Unsupported Data for that command.` (Could be an issue with my model).

Reply:
The statistics command is a more recent addition (I year ago?).
I read about it in a blog entry.
You have to update your firmware to get that feature.
The wear is relative to the last calibration **after** the firmware update (I'd say by comparing the calibration data 😁 ).

For the firmware update you can use their software: https://vacugraph.com/

Attention: The firmware update might change the baud_rate (in my case from 9600 to 115200), I had to use a port sniffer to detect that change, as the official software continued to work.

Reply:
thanks for the hint. It seems I have the latest firmware already. but the hot cathode (in contrast to what is written in the documentation) seems to not support this command. Also the Vacugraph software shows the wear data only for the Pirani channel. For me this feature is not so crucial.